filesystems/gcs/src/main/scala/cromwell/filesystems/gcs/GcsPathBuilder.scala

```diff
@@ -20,65 +19,77 @@ import cromwell.filesystems.gcs.auth.GoogleAuthMode
 import scala.concurrent.duration._
 import scala.concurrent.{ExecutionContext, Future}
 import scala.language.postfixOps
-import scala.util.Try
+import scala.util.{Failure, Try}
 
 object GcsPathBuilder {
 
   val JsonFactory = JacksonFactory.getDefaultInstance
   val HttpTransport = GoogleNetHttpTransport.newTrustedTransport
-
-  def checkValid(uri: URI) = {
-    checkNotNull(uri.getScheme, s"%s does not have a gcs scheme", uri)
-    checkArgument(
-      uri.getScheme.equalsIgnoreCase(CloudStorageFileSystem.URI_SCHEME),
-      "Cloud Storage URIs must have '%s' scheme: %s",
-      CloudStorageFileSystem.URI_SCHEME: Any,
-      uri: Any
-    )
-    checkNotNull(uri.getHost, s"%s does not have a host", uri)
-  }
+  // Provides some level of validation of GCS bucket names
+  // This is meant to alert the user early if they mistyped a gcs path in their workflow / inputs and not to validate
+  // exact bucket syntax, which is done by GCS.
+  // See https://cloud.google.com/storage/docs/naming for full spec
+  val GcsBucketPattern =
+    """
+      (?x)                                      # Turn on comments and whitespace insensitivity
+      gs://
```

Comment:
does this need a `^` anchor on the front in case this pattern occurs internally?